internal/schema/typeref.go

```diff
@@ -119,6 +107,65 @@ func (r *TypeRef) IsScalarID() bool {
 	return r.OfType != nil && r.OfType.Kind == KindScalar && r.GetTypeName() == "ID"
 }
 
+// IsNonNull walks down looking for NON_NULL kind, however that can appear
+// multiple times, so this is likely a bit deceiving...
+// Example:
+// {
+//    "name": "tags",
+//    "description": "An array of key-values pairs to represent a tag. For example:  Team:TeamName.",
+//    "type": {
+//     "kind": "NON_NULL",
+//     "ofType": {
+//      "kind": "LIST",
+//      "ofType": {
+//       "kind": "NON_NULL",
+//       "ofType": {
+//        "name": "TaggingTagInput",
+//        "kind": "INPUT_OBJECT"
+//       }
+//      }
+//     }
+//    }
+//   }
+//  ]
+// }
 func (r *TypeRef) IsNonNull() bool {
-	return r.Kind == KindNonNull
+	kinds := r.GetKinds()
+
+	// Lots of kinds
+	for _, k := range kinds {
+		if k == KindNonNull {
+			return true
+		}
+	}
+
+	return false
+}
+
+// IsList determines if a TypeRef is of a KIND LIST.
+func (r *TypeRef) IsList() bool {
```

Comment:
I think for this to be accurate, its only the first kind that is a list, which would make the TypeRef a list.

Reply:
If a List is Non-null, then then it's not the first Kind, it would be at least the second.  I've seen this pattern more than once:

`NON_NULL, LIST, NON_NULL, OBJECT`